Various improvements related to repos/snapshots/publishes

bbinet opened the pull request:
And also add support for more options when publishing a new snapshot.

(backward compatibility with existing pillar structure is preserved)

bbinet commented:
Please wait before merging this PR.
I will extract the new snapshot publishing capability to a new aptly/server/publish.sls state so that it is not related to mirrors anymore (and can be used to publish repo snapshots too).

Any objection before I start the refactoring?

fpytloun commented:
I am fine with it, go ahead :-)
Only two notes:
- it's recommended to use aptly-publisher for creating publishes as that tool is idempotent (won't do anything if there's nothing to do)
- I am planning to add feature to create up-to-date snapshots when publishing with aptly-publisher so it will not be needed to create them by other tools

bbinet commented:
Yes, but as discussed in the other PR, I was not able to publish my filtered mirrors with aptly-publisher (probably because of an aptly api bug), so I'm now using salt directly, trying to make the salt formula publishing part idempotent too.

fpytloun commented:
Yes, I am fine with using it this way.

bbinet committed:
Add support for dropping mirror snapshots

bbinet committed:
Fix requirement issue when mirror.update is false

bbinet committed:
Fix missing requirement when engine is not docker

bbinet committed:
Minor renaming simplification

bbinet committed:
Add support for snapshotting repos

bbinet committed:
Add support for publishing snapshots

(works for both mirrors and repo snapshots)

bbinet committed:
Only run aptly repo add if packages are available in pkg_dir

bbinet committed:
Add support for aptly repo add -remove-files option

bbinet committed:
Add pillar example in readme for aptly server publish

bbinet commented:
@fpytloun I've updated the PR with the refactoring and some other improvements:
- 5891a0c Add support for dropping mirror snapshots
- 48360ca Fix requirement issue when mirror.update is false
- 4e6621a Fix missing requirement when engine is not docker
- eb60e61 Minor renaming simplification
- f3f9369 Add support for snapshotting repos
- 1430653 Add support for publishing snapshots
- fde4d3c Only run aptly repo add if packages are available in pkg_dir
- ec80056 Add support for aptly repo add -remove-files option
- f4610cb Add pillar example in readme for aptly server publish

Every commit include a specific self-contained feature so that it is easier to review.

bbinet committed:
Add support for aptly repo include

to be able to import .changes files (from dpkg-genchanges)
This also add support for importing gpg keys to verify .changes
signature

bbinet committed:
Fix pillar example for gpgkeys syntax in readme

(must be an array)

bbinet committed:
Cosmetics: cleanup .get('var', None) syntax

=> can be written: .get('var')

bbinet commented:
I've updated the pull request with a new feature to add support for aptly repo include to be able to import `.changes` files (from `dpkg-genchanges`).
This also add support for importing gpg keys to verify `.changes` signature.

So here are now the various improvements of this PR:
- 5891a0c Add support for dropping mirror snapshots
- 48360ca Fix requirement issue when mirror.update is false
- 4e6621a Fix missing requirement when engine is not docker
- eb60e61 Minor renaming simplification
- f3f9369 Add support for snapshotting repos
- 1430653 Add support for publishing snapshots
- fde4d3c Only run aptly repo add if packages are available in pkg_dir
- 9e68896 Add support for aptly repo add -remove-files option
- 35af213 Add pillar example in readme for aptly server publish
- a7521cb Add support for aptly repo include
- 5e9c82b Fix pillar example for gpgkeys syntax in readme
- ce54f1d Cosmetics: cleanup .get('var', None) syntax

This PR is now ready to be reviewed.

bbinet commented:
@fpytloun do you think this can be merged soon?

bbinet committed:
Fix conflicting salt ids when managing multiple repos

bbinet commented:
@fpytloun any news on this one?

bbinet committed:
Python3 compatibility: use items() instead of iteritems()
